### PR TITLE
Add support for mobile UI filtering

### PR DESCRIPTION
**Pull request by bryce-b**
## Motivation/summary
The mobile ui wants to filter data based on host.os.version & service.version. Http spans are a key part of the mobile ui, and when filters are added for either of these attributes all http related charts will be empty.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

https://github.com/elastic/apm-agent-ios/issues/145

### CODE REVIEW

**Review by axw — COMMENTED** [review]
@felixbarny do you think this is worth benchmarking, or would the design win anyway?

I wonder if we should drop that `remove` processor altogether. This has been around since forever, when we used to store a transaction document even for non-sampled transactions, and when we didn't have a default ILM delete phase . I feel like we have better options for storage management these days.

**Comment by felixbarny** [comment]
If we already had synthetic _source, it would probably be a no-brainer to drop the `remove` processor altogether. Instinctively, I don't expect this to have a huge impact even now but I'd prefer to do some testing for that. If that testing would be very difficult to do (which I hope it isn't, that would kinda be another issue in itself) I suppose we could skip it if we preserve just `host` and `service.name`.

**Review by axw — APPROVED** [review]

File: changelogs/head.asciidoc [hunk 1]
```diff
@@ -17,3 +17,4 @@ https://github.com/elastic/apm-server/compare/8.8\...main[View commits]
 
 [float]
 ==== Added
+- Span events now carry `host` and `service.version` attributes.
```
Comment (kruskall) [inline changelogs/head.asciidoc:20]:
```suggestion
- Span events now carry `host` and `service.version` attributes. {pull}10697[10697]
```

**Review by kruskall — CHANGES_REQUESTED** [review]

**Comment by axw** [comment]
Verified with BC2. Send testdata/intake-v2/spans.ndjson to the hosted APM Server, confirmed that `host.*` and `service.version` fields are present in span documents.

**Pull request merged**